examples/json-record-map-conversion/json_record_map_conversion.bal

```diff
@@ -28,16 +32,16 @@ public function main() {
     // This example shows how you can convert a record to a JSON object.
     // This conversion could return an error because it may not be possible
     // to convert some data types that are defined in the record to JSON.
-    json|error j = json.constructFrom(theRevenant);
+    json|error j = theRevenant.cloneWithType(j1);
```

Comment:
```suggestion
    json|error j = theRevenant.cloneWithType(json);
```
Wouldn't this work?

Reply:
This should work. You'll only have to define a type for types like unions.